static/js/formatters-internal.js

```diff
@@ -82,15 +82,23 @@ export function toLocalizedDistance(profile, key = 'd_distance', displayUnits) {
   return this.toMiles(profile, undefined, undefined, locale);
 }
 
+export function _getDefinedLocale(locale) {
+  return _getJSFormattedLocale(locale) || _getDocumentLocale() || 'en';
+}
+
+export function _getJSFormattedLocale(locale) {
+  return locale && locale.replace(/_/g, '-');
+}
+
 export function _getDocumentLocale() {
-  return document.documentElement.lang.replace('_', '-');
+  return _getJSFormattedLocale(document.documentElement.lang);
```

Comment:
can we just update this to use parseLocale?

Reply:
parseLocale provide an object of different section of the locale, and we would have to piece them together with dash afterward, which would give us the same result. Is there another reason why you think we should use parseLocale here?

Reply:
For some reason I thought parseLocale would canonicalize stuff for us. I'm still a little confused on the difference between the 3 different methods we have here though

Reply:
`_getJSFormattedLocale` is just to replace underscore with dashes in a **given locale string** as of now because JS functions like toLocaleString and Int.NumberFormat would throw an error otherwise. `_getDocumentLocale()` is to provide a JS formatted locale from the **document**. In a lot of places we would go through the logic of using the given locale if that exist or use the document locale or fallback to en as last resort, so I added `_getDefinedLocale` to avoid duplicating that code, which return a defined locale in JS format.

Reply:
could we just use the same locale everywhere? it feels like a potential source of confusion where some formatters use different flavors of the same locale. Is it necessary that some formatters directly use the documents locale and others do some preprocessing first?

Reply:
like it feels like this introduces a lot of entropy that we might not need

Reply:
getDefinedLocale makes it more difficult for me to reason about where the locale came from and what format it is. Another approach would be to create a helper such as `formatLocaleForIntl` which replaces the underscores with dashes so that it doesn't break when passed to Intl. The line to get the locale would then be `locale = formatLocaleForIntl(locale || _getDocumentLocale())`

Reply:
Another change we could make which isn't strictly necessary is to also return a full canonicalized locale from parseLocale

Reply:
Maybe renaming '_getJSFormattedLocale' to '_getLocaleWithDashes' will be clearer? Otherwise, PR looks good